### PR TITLE
feat: add support to add parameters in request to push view

### PR DESCRIPTION
**Pull request by uziasferreirazup**
### Related Issues

#1398

### Description and Example

the android module was removed to have only one request object for the entire application, with that we will have a consistency in the application

### Checklist

Please, check if these important points are met using `[x]`:

- [X] I read the [PR Guide] and followed the process outlined there for submitting this PR.
- [X] I avoided _breaking changes_ by not changing public APIs that people rely on. 
- [X] I am willing to follow-up on review comments in a timely manner.

[PR Guide]: https://github.com/ZupIT/beagle/blob/master/doc/contributing/pull_requests.md

### CODE REVIEW

**Comment by botbeaglezup** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">The title of PR not follow the rules of: https://www.conventionalcommits.org/en/v1.0.0/</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: android/beagle/src/main/java/br/com/zup/beagle/android/data/ComponentRequester.kt [hunk 1]
```diff
@@ -19,26 +19,24 @@ package br.com.zup.beagle.android.data
 import br.com.zup.beagle.android.cache.CacheManager
 import br.com.zup.beagle.android.data.serializer.BeagleSerializer
 import br.com.zup.beagle.android.exception.BeagleException
-import br.com.zup.beagle.android.view.ScreenRequest
-import br.com.zup.beagle.android.view.mapper.toRequestData
+import br.com.zup.beagle.android.networking.RequestData
 import br.com.zup.beagle.core.ServerDrivenComponent
```
Comment (botbeaglezup) [inline android/beagle/src/main/java/br/com/zup/beagle/android/data/ComponentRequester.kt:23]: <table data-meta="generated_by_android">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/12e93bb033e1ea5a1bc1540056cde259b3dec607/android/beagle/src/main/java/br/com/zup/beagle/android/data/ComponentRequester.kt#L23"></span>The import 'br.com.zup.beagle.android.view.ScreenRequest' is unused.</td>
    </tr>
  </tbody>
</table>

File: android/beagle/src/main/java/br/com/zup/beagle/android/data/ComponentRequester.kt [hunk 2]
```diff
@@ -19,26 +19,24 @@ package br.com.zup.beagle.android.data
...
 import kotlin.jvm.Throws
```
Comment (botbeaglezup) [inline android/beagle/src/main/java/br/com/zup/beagle/android/data/ComponentRequester.kt:24]: <table data-meta="generated_by_android">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/12e93bb033e1ea5a1bc1540056cde259b3dec607/android/beagle/src/main/java/br/com/zup/beagle/android/data/ComponentRequester.kt#L24"></span>The import 'br.com.zup.beagle.android.view.mapper.toRequestData' is unused.</td>
    </tr>
  </tbody>
</table>

File: android/beagle/src/main/java/br/com/zup/beagle/android/utils/ViewGroupExtensions.kt [hunk 3]
```diff
@@ -14,6 +14,8 @@
  * limitations under the License.
  */
 
+@file:Suppress("TooManyFunctions")
```
Comment (botbeaglezup) [inline android/beagle/src/main/java/br/com/zup/beagle/android/utils/ViewGroupExtensions.kt:17]: <table data-meta="generated_by_android">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/12e93bb033e1ea5a1bc1540056cde259b3dec607/android/beagle/src/main/java/br/com/zup/beagle/android/utils/ViewGroupExtensions.kt#L17"></span>File '/Users/runner/work/beagle/beagle/android/beagle/src/main/java/br/com/zup/beagle/android/utils/ViewGroupExtensions.kt' with '13' functions detected. Defined threshold inside files is set to '13'</td>
    </tr>
  </tbody>
</table>

File: android/beagle/src/main/java/br/com/zup/beagle/android/utils/ViewGroupExtensions.kt [hunk 4]
```diff
@@ -79,6 +168,10 @@ fun ViewGroup.loadView(
  * @property screenRequest to create your request data to fetch the component
  * @property listener is called when the loading is Started, Finished and Success
  */
+@Deprecated("This method was deprecated in version 1.7.0 and will be removed in a future version." +
```
Comment (botbeaglezup) [inline android/beagle/src/main/java/br/com/zup/beagle/android/utils/ViewGroupExtensions.kt:171]: <table data-meta="generated_by_android">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/12e93bb033e1ea5a1bc1540056cde259b3dec607/android/beagle/src/main/java/br/com/zup/beagle/android/utils/ViewGroupExtensions.kt#L171"></span>Line detected that is longer than the defined maximum line length in the code style.</td>
    </tr>
  </tbody>
</table>

File: android/beagle/src/main/java/br/com/zup/beagle/android/utils/ViewGroupExtensions.kt [hunk 5]
```diff
@@ -101,14 +194,18 @@ fun ViewGroup.loadView(
  * @property listener is called when the loading is Started, Finished and Success
  */
 @JvmName("loadView2")
```
Comment (botbeaglezup) [inline android/beagle/src/main/java/br/com/zup/beagle/android/utils/ViewGroupExtensions.kt:196]: <table data-meta="generated_by_android">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/12e93bb033e1ea5a1bc1540056cde259b3dec607/android/beagle/src/main/java/br/com/zup/beagle/android/utils/ViewGroupExtensions.kt#L196"></span>Line detected that is longer than the defined maximum line length in the code style.</td>
    </tr>
  </tbody>
</table>

File: android/beagle/src/main/java/br/com/zup/beagle/android/view/BeagleActivity.kt [hunk 6]
```diff
@@ -32,13 +32,16 @@ import androidx.lifecycle.ViewModelProvider
 import br.com.zup.beagle.R
 import br.com.zup.beagle.android.components.layout.Screen
 import br.com.zup.beagle.android.data.serializer.BeagleSerializer
+import br.com.zup.beagle.android.networking.RequestData
 import br.com.zup.beagle.android.setup.BeagleEnvironment
```
Comment (botbeaglezup) [inline android/beagle/src/main/java/br/com/zup/beagle/android/view/BeagleActivity.kt:36]: <table data-meta="generated_by_android">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/12e93bb033e1ea5a1bc1540056cde259b3dec607/android/beagle/src/main/java/br/com/zup/beagle/android/view/BeagleActivity.kt#L36"></span>The import 'br.com.zup.beagle.android.networking.HttpAdditionalData' is unused.</td>
    </tr>
  </tbody>
</table>

File: android/beagle/src/main/java/br/com/zup/beagle/android/view/custom/BeagleView.kt [hunk 7]
```diff
@@ -18,16 +18,18 @@ package br.com.zup.beagle.android.view.custom
 
 import android.annotation.SuppressLint
 import android.view.View
-import androidx.lifecycle.Observer
+import br.com.zup.beagle.android.networking.RequestData
```
Comment (botbeaglezup) [inline android/beagle/src/main/java/br/com/zup/beagle/android/view/custom/BeagleView.kt:21]: <table data-meta="generated_by_android">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/12e93bb033e1ea5a1bc1540056cde259b3dec607/android/beagle/src/main/java/br/com/zup/beagle/android/view/custom/BeagleView.kt#L21"></span>The import 'androidx.lifecycle.Observer' is unused.</td>
    </tr>
  </tbody>
</table>

**Comment by botbeaglezup** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/ZupIT/beagle/blob/12e93bb033e1ea5a1bc1540056cde259b3dec607/android/beagle/src/main/java/br/com/zup/beagle/android/view/BeagleActivity.kt#L21">android/beagle/src/main/java/br/com/zup/beagle/android/view/BeagleActivity.kt#L21</a> - The import 'android.net.Uri' is unused.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_android">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: backend/widgets/src/main/kotlin/br/com/zup/beagle/widget/networking/HttpAdditionalData.kt [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+ * limitations under the License.
+ */
+
+package br.com.zup.beagle.widget.networking
+
+/**
+ * HttpAdditionalData is used to do requests.
+ * @param method HTTP method.
+ * @param headers Header items for the request.
+ * @param body Content that will be deliver with the request.
```
Comment (matheusribeirozup) [inline backend/widgets/src/main/kotlin/br/com/zup/beagle/widget/networking/HttpAdditionalData.kt:23]:
```suggestion
 * @param body Content that will be delivered with the request.
```

**Review by paulomeurerzup — DISMISSED** [review]

**Review by matheusribeirozup — APPROVED** [review]

**Review by viniciusguardieirozup — APPROVED** [review]

**Pull request merged**